packages/caravan-psbt/README.md

```diff
@@ -104,6 +108,54 @@ An object class representing a PSBTv2 and its current state. While not yet compl
 
 Getters and setters are provided for the keytypes defined in [BIP 174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki) and [BIP 370](https://github.com/bitcoin/bips/blob/master/bip-0370.mediawiki). These getters/setters are named after the keytype in ALL_CAPS_SNAKE like `PsbtV2.PSBT_GLOBAL_VERSION`. Additional getters/setters may be listed in the following API documentation.
 
+##### `get isReadyForConstructor`
+
+A getter to check readiness for an operator role. Operator roles are defined in defined in [BIP 174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki#user-content-Roles) and [BIP 370](https://github.com/bitcoin/bips/blob/master/bip-0370.mediawiki#user-content-Roles).
+
+Returns `true` if the PsbtV2 is ready for an operator taking the Constructor role.
```

Comment:
The Constructor is responsible for adding inputs and outputs. The psbt is not in a Constructor role, but it can be ready for a Constructor as long as it's still possible to add inputs or outputs to the psbt. This getter checks to see if at least inputs or outputs are still modifiable.

When a signature is added to the psbt by `addPartialSig`, the sighash is considered and `handleSighashType` updates the inputs/outputs modifiable. It's likely that a signature would remove both modifiable flags. In that case, the psbt is no longer ready for a Constructor.